applications/nrf5340_audio/src/modules/audio_usb.c

```diff
@@ -40,71 +55,132 @@ struct audio_metadata usb_in_meta = {.data_coding = PCM,
 						  BT_AUDIO_LOCATION_FRONT_RIGHT,
 				     .bad_data = 0};
 
-#if (CONFIG_STREAM_BIDIRECTIONAL)
 static uint32_t tx_num_underruns;
 
-static void data_write(const struct device *dev)
+static void terminal_update_cb(const struct device *dev, uint8_t terminal, bool enabled,
+			       bool microframes, void *user_data)
+{
+	ARG_UNUSED(dev);
+	ARG_UNUSED(terminal);
+	ARG_UNUSED(microframes);
+	ARG_UNUSED(user_data);
+
+	terminal_enabled = enabled;
+}
+
+static void usb_send_cb(const struct device *dev, void *user_data)
 {
 	int ret;
+	void *pcm_buf;
+	static struct net_buf *data_out;
 
-	if (audio_q_tx == NULL) {
-		LOG_WRN("USB TX queue not initialized, dropping data.");
+	if (audio_q_tx == NULL || !terminal_enabled) {
```

Comment:
consider a LOG_RATELIMIT here

Reply:
Terminal is not yet enabled, so no need to print anything